include/jemalloc/internal/pages.h

```diff
@@ -27,6 +27,9 @@ extern size_t	os_page;
 #define HUGEPAGE	((size_t)(1U << LG_HUGEPAGE))
 #define HUGEPAGE_MASK	((size_t)(HUGEPAGE - 1))
 
+/* Used to validate that the hugepage size is not unexpectedly high. */
+#define HUGEPAGE_MAX_EXPECTED_SIZE ((size_t)(16U << 20))
```

Comment:
The value is picked based on the info here: https://wiki.debian.org/Hugepages#Huge_pages_sizes
```
Architecture	huge page size
arm64	4K, 2M and 1G (or 64K and 512M if one builds their own kernel with CONFIG_ARM64_64K_PAGES=y)
i386	4K and 4M (2M in PAE mode)
ia64	4K, 8K, 64K, 256K, 1M, 4M, 16M, 256M
ppc64	4K and 16M
```
as it looks the values beyond 16M are all too large and not expected.

Reply:
Maybe also elaborate in the comments why too high a huge-page size is harmful?

Reply:
Makes sense. Added more details.